03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,29 +1,48 @@
 
 const Task = require('../models/Task')
-
-const createTask = async(req,res) => {
+const asyncWrapper = require('../middleware/async')
+const {createCustomError} = require('../errors/custom-error')
+const getAllTasks = asyncWrapper (async (req,res) => {
+    const tasks = await Task.find({})
+    res.status(200).json({tasks})
+})
+
+const createTask = asyncWrapper ( async(req,res) => {
     const task = await Task.create(req.body)
     res.status(201).json({ task }) 
+})
+
+const getTask =  asyncWrapper (async(req,res, next) => {
+    const {id: taskID }= req.params
+    const task = await Task.findOne({_id:taskID})
+    if(!task){
+        return next(createCustomError(`No task with id : ${taskID}`, 404))
+    }
+    res.status(200).json({task})
+})
+
+
+const deleteTask =  asyncWrapper (async (req,res) => {
+    const {id: taskID} = req.params
+    const task = await Task.findOneAndDelete({_id:taskID})
+    if(!task){
+        return res.status(404).json({msg: `No task with id : ${taskID}`})
+    }
+    res.status(200).json({task})
+})
+
+const updateTask =  asyncWrapper (async (req,res) => {
+        const {id:taskID} = req.params; 
+        const task = await Task.findOneAndUpdate({_id:taskID}, req.body, {new: true,
+             runValidators: true,
+            overwrite:true,
+        })
+        if(!task){
+            return next(createCustomError(`No task with id : ${taskID}`, 404))
+        }
+        res.status(200).json({task})
 }
-
-const getTask = (req,res) => {
-    res.json({id:req.params.id})
-}
-
-const getAllTasks = (req,res) => {
-    res.send('get all tasks')
-}
-
-
-const updateTask = (req,res) => {
-    res.send('update task')
-}
-
-const deleteTask = (req,res) => {
-    res.send('delete task ')
-}
-
-
+)
 
 module.exports = {
     getAllTasks, createTask, getTask,
```

Comment:
The module.exports block isn't completed.  create, get, delete, update should all be exported. and it should be closed with }